test/integration/targets/var_templating/runme.sh

```diff
@@ -14,4 +14,5 @@ ansible-playbook undall.yml -i inventory -v "$@"
 # test hostvars templating
 ansible-playbook task_vars_templating.yml -v "$@"
 
-ansible-playbook test_connection_vars.yml -v "$@" 2>&1 | grep 'sudo'
+# there should be an attempt to use 'sudo' in the connection debug output
+ansible-playbook test_connection_vars.yml -vvvv "$@" | tee /dev/stderr | grep 'sudo'
```

Comment:
I don't think this is doing what you want though.  If you run this, and look at the match, it's matching `become_method: sudo` from:

```
PLAYBOOK: test_connection_vars.yml *********************************************
Positional arguments: test_connection_vars.yml
force_handlers: True
become_method: sudo
inventory: (u'/dev/null',)
forks: 5
tags: (u'all',)
verbosity: 4
connection: smart
timeout: 10
1 plays in test_connection_vars.yml
```

Those are just the CLI invocation details, and not the play configuration.

Because the test runs as root, `become` will be ignored, unless you also specify `ANSIBLE_BECOME_ALLOW_SAME_USER=1`

In which case the `grep` also needs to be updated to match the command that is executed, in a way that doesn't match the CLI invocation.  Like grepping for `sudo \-H` or something.